agent/functional_tests/generators/simpletests.go

```diff
@@ -63,44 +63,78 @@ import (
 func Test{{ $el.Name }}(t *testing.T) {
+	{{if $el.DockerVersion}}
+	// Test only available for docker version {{ $el.DockerVersion }}
+	RequireDockerVersion(t, "{{ $el.DockerVersion }}") 
+	{{end}}
 	// Parallel is opt in because resource constraints could cause test failures
 	// on smaller instances
 	if os.Getenv("ECS_FUNCTIONAL_PARALLEL") != "" { t.Parallel() }
 	agent := RunAgent(t, nil)
 	defer agent.Cleanup()
 	agent.RequireVersion("{{ $el.Version }}")
 
-	testTask, err := agent.StartTask(t, "{{ $el.TaskDefinition }}")
+	td, err := GetTaskDefinition("{{ $el.TaskDefinition }}")
 	if err != nil {
-		t.Fatal("Could not start task", err)
+		t.Fatalf("Could not register task definition: %%v", err)
 	}
-	timeout, err := time.ParseDuration("{{ $el.Timeout }}")
+	testTasks, err := agent.StartMultipleTasks(t, td, {{ $el.Count }})
 	if err != nil {
-		t.Fatal("Could not parse timeout", err)
+		t.Fatalf("Could not start task: %%v", err)
 	}
-	err = testTask.WaitStopped(timeout)
+	timeout, err := time.ParseDuration("{{ $el.Timeout }}")
 	if err != nil {
-		t.Fatalf("Timed out waiting for task to reach stopped. Error %%#v, task %%#v", err, testTask)
+		t.Fatalf("Could not parse timeout: %%#v", err)
 	}
 
-	{{ range $name, $code := $el.ExitCodes }}
-	if exit, ok := testTask.ContainerExitcode("{{$name}}"); !ok || exit != {{ $code }} {
-		t.Errorf("Expected {{$name}} to exit with {{$code}}; actually exited (%%v) with %%v", ok, exit)
+	{{if $el.LongRunningTask}}
+	// Make sure the task is running
+	for _, testTask := range testTasks {
+		err = testTask.WaitRunning(timeout)
+		if err != nil {
+			t.Errorf("Timed out waiting for task to reach running. Error %%v, task %%v", err, testTask)
+		}
+	}
+
+	// Cleanup, stop all the tasks and wait for the containers to be stopped
+	for _, testTask := range testTasks {
+		err = testTask.Stop()
+		if err != nil {
+			t.Errorf("Failed to stop task, Error %%v, task %%v", err, testTask)
+		}
 	}
-	{{ end }}
+	{{end}}
+
+	for _, testTask := range testTasks {
+		err = testTask.WaitStopped(timeout)
+		if err != nil {
+			t.Fatalf("Timed out waiting for task to reach stopped. Error %%#v, task %%#v", err, testTask)
+		}
+
+		{{ range $name, $code := $el.ExitCodes }}
+		if exit, ok := testTask.ContainerExitcode("{{$name}}"); !ok || exit != {{ $code }} {
+			t.Errorf("Expected {{$name}} to exit with {{$code}}; actually exited (%%v) with %%v", ok, exit)
+		}
+		{{ end }}
+		defer agent.SweepTask(testTask)
+	}
+
 }
 {{ end }}
 `
 
 func main() {
 	type simpleTestMetadata struct {
-		Name           string
-		Description    string
-		TaskDefinition string
-		Timeout        string
-		ExitCodes      map[string]int
-		Tags           []string
-		Version        string
+		Name            string
+		Description     string
+		TaskDefinition  string
+		Timeout         string
+		ExitCodes       map[string]int
+		Tags            []string
+		Version         string
+		Count           int
+		DockerVersion   string
+		LongRunningTask bool
```

Comment:
I wonder if there's a better name for this...maybe `Daemon` since that's a typical way to refer to a process that runs indefinitely?  Or maybe just `IndefinitelyRunningTask`?  I think `LongRunningTask` implies that it could still be finite in length.